Feature/37108 improve design of GitHub tab

wielinde opened the pull request:
https://community.openproject.org/wp/37108

wielinde committed:
Rename Git button to 'Git snippets'

wielinde committed:
Change font for Github tab header

wielinde committed:
Improve design of pr status badges

wielinde committed:
Reorganize the design of PR headers

wielinde committed:
Improve design of PR checks

wielinde committed:
Improve design of git snippets menu

wielinde committed:
Remove WP description from commit message and command

wielinde committed:
Fix frontend specs for github integration

oliverguenther commented:
I spawned a PP instance with an integration to https://github.com/opf/qa-integration-test/ and a few tests (e.g., https://pr-9329-feature-37108-improv-ip-52-29-42-228.my.pullpreview.com/projects/demo-project/work_packages/details/34/github) 

Mostly LGTM 👍 

- [ ] The details link in the github actions should be a `_target=blank` as it's an external link and we usually force a new window for that.
- [ ] Compared to the WP author line, there is a dot missing after the author name in the github tab.
<img width="473" alt="Bildschirmfoto 2021-05-28 um 21 48 51" src="https://user-images.githubusercontent.com/459462/120035252-9063f380-bffe-11eb-8c61-07f564aaf6ca.png">

- [ ] The status pill is too large compared to the title. I remember the title being bigger in the figma mockup?
- [ ] There's a line after each PR entry. This wasn't there in Figma but I don't mind itself. However you should omit the last one. 
<img width="565" alt="Task__Github_PR___34____Demo_project___OpenProject" src="https://user-images.githubusercontent.com/459462/120035562-1c761b00-bfff-11eb-81f7-6a11c03a50cf.png">

wielinde committed:
Adding dots to the end of phrases.

wielinde committed:
Github Integration: reduce dom depth. remove last border under PR

wielinde committed:
Add target blank to PR details link & other fixes

wielinde committed:
Fix work_package_github_tab_spec.rb

ulferts commented on modules/github_integration/frontend/module/git-actions-menu/git-actions-menu.component.spec.ts:
```diff
@@ -11,7 +11,7 @@ import { By } from "@angular/platform-browser";
 
 
-describe('GitActionsMenuComponent', () => {
+fdescribe('GitActionsMenuComponent', () => {
```
`fdescribe` relict

wielinde replied:
Oops, thanks!

wielinde committed:
Remove fdescribe

wielinde commented:
- [x] Added `target="blank"` to the "Details" link.
- [x] Created real phrases with dots at the end for "Created by" and "Last updatede on".
- [x] Removed the border-bottom from the last PR in the list... For that I had to change (reduce) the DOM structure a bit and use the `host`.
- [x] Fixed specs

@oliverguenther Yes, I diverged a bit from the Figma prototype as Figma's context was different. For example the Tab labels were designed differently. So I free-styled here a bit. I didn't want to make the PR titles bigger as the main headline of that page is still the WP subject itself. The PRs are of secondary importance. You find the PR status pill too big. Call me blind (betriebsblind) but I already don't see an issue here.

wielinde commented:
Ooops, something is broken, now... I can't close the "Git snippets" menu. I'll check that.

ulferts commented:
I also like almost all the changes. Does look way better now. 

I am not satisfied with the header, though:

![image](https://user-images.githubusercontent.com/617519/120193860-31c89080-c21d-11eb-8dea-7decf59bf9f8.png)

To me, the distance between "Pull Requests" and the line below is too big. Because of this, the style differs from: 

The attribute group & relations header:

![image](https://user-images.githubusercontent.com/617519/120194261-aac7e800-c21d-11eb-93b1-d4036da4ecd0.png)

![image](https://user-images.githubusercontent.com/617519/120194349-caf7a700-c21d-11eb-883a-f8f224314a12.png)

The activity header: 

![image](https://user-images.githubusercontent.com/617519/120194418-dfd43a80-c21d-11eb-987a-2279346c29a3.png)

Can we get the github header to look like the attribute group/relations header?

wielinde committed:
Improve layout of Github pull requests tab header

wielinde committed:
Add title to Github check icon

Commit:
Refactor classnames for github pull requests

This introduces `op-pull-request` and `op-pr-check` as BEM blocks, both in their own sass files, but the files still scoped to the pull request component.

wielinde committed:
Fix frontend spec. Added karma Chrome debug config.

wielinde committed:
Fix frontend specs after refactoring styles
